### PR TITLE
Enhance Installation Script with Robust Error Handling and Logging

### LINKED ISSUE #2028: Reimagining the Installation Process for Tempesta FW
I would like to propose a review of our current approach to product installation, considering issues #2026 and #2027. Perhaps it makes sense to make changes to the installation process.

I suggest considering the option of checking the environment for compliance with the product's system requirements before starting the installation. This will allow us to make informed decisions regarding the possibility of installing the product and will improve the experience for end users.

Additionally, it's worth discussing the necessity of including a DKMS module in the installer, given that currently only one kernel is supported.

### PR DESCRIPTION
**Pull request by ykargin**
Fix #2028 

- Improved error handling and logging mechanisms throughout the installation script.
- Added user prompts and important notices to improve user interaction and clarify the installation process.
- Implemented robust argument parsing with error checks to handle invalid input gracefully.
- Refined DKMS module installation procedure to alert users about potential error messages for unsupported kernels and ensure a smoother installation experience.
- Adjusted regular expressions for package detection to accurately match specific kernel versions.

### CODE REVIEW

File: pkg/scripts/tempesta_installer.sh [hunk 1]
```diff
@@ -18,29 +18,34 @@
 # this program; if not, write to the Free Software Foundation, Inc., 59
```
Comment (krizhanovsky) [inline pkg/scripts/tempesta_installer.sh:18]: Please update the copyright year or (better) test the recent Anton's github hook that it does update the copyright

File: pkg/scripts/tempesta_installer.sh [hunk 2]
```diff
@@ -170,52 +202,101 @@ tfw_install_deps()
...
 	# TempestaFW is shipped as DKMS module. By default, dkms framework
 	# installs module only for running kernel. Configure dkms to install
 	# module to all kernels. BUILD_EXCLUSIVE variable in TempestaFW's
 	# dkms.conf will prevent building against unsupported kernels.
+	log "INFO" "Configuring DKMS for all kernels."
 	echo -e "autoinstall_all_kernels=\"yes\"" >> /etc/dkms/framework.conf
-
+	echo ""
+	echo "*************************************************************************"
+	echo "* IMPORTANT: TempestaFW is distributed as a DKMS module.                *"
```
[the remark this answers is not in the document]
Comment (ykargin) [reply pkg/scripts/tempesta_installer.sh:235]: > Please create a task for 0.9 about movement to binary packages

Done: #2038

File: pkg/scripts/tempesta_installer.sh (hunk 1 above)
Comment (ykargin) [reply pkg/scripts/tempesta_installer.sh:18]: > Please update the copyright year or (better) test the recent Anton's github hook that it does update the copyright

Done (by hook)

**Review by RomanBelozerov — APPROVED** [review]

**Review by const-t — APPROVED** [review]
LGTM

**Pull request merged**